projects/generate_glitches/generate_glitches.py

```diff
@@ -0,0 +1,342 @@
+import glob
+import logging
+import os
+from typing import Optional
+
+import gwdatafind
+import h5py
+import numpy as np
+from gwpy.segments import Segment, SegmentList
+from gwpy.timeseries import TimeSeries
+from hermes.typeo import typeo
+from tqdm import tqdm
+
+"""
+Tools to generate a dataset of glitches from omicron triggers.
+
+For information on how the omicron triggers were generated see:
+
+/home/ethan.marx/bbhnet/generate-glitch-dataset/omicron/12566/H1L1_1256665618_100000
+/runfiles/omicron_params_H1.txt
+
+on CIT cluster for an example omicron parameter file.
+
+Of note is the clustering timescale of 1 second
+
+"""
+
+
+def veto(times: list, segmentlist: SegmentList):
+
+    """
+    Remove events from a list of times based on a segmentlist
+    A time ``t`` will be vetoed if ``start <= t <= end`` for any veto
+    segment in the list.
+
+    Arguments:
+    - times: the times of event triggers to veto
+    - segmentliss: the list of veto segments to use
+
+    Returns:
+    - keep_bools: list of booleans; True for the triggers to keep
+    """
+
```

Comment:
There's actually a pretty fun array-like way of doing this that I just tested if you're interested:
```
    mask = True
    for t0, tf in segmentlist:
        mask &= (t0 >= times) | (times >= tf)
    return mask
```

Reply:
@alecgunny Wow much simpler. I could be wrong but I think for vetoes files with many more segments this may take longer since each segment is checked every time for every trigger. Will keep this in mind though.